src/ui/simple_ui/ask_hash.rs

```diff
@@ -55,6 +98,35 @@ pub fn ask_hash(args: &BurnArgs, cf: CompressionFormat) -> anyhow::Result<Option
     Ok(Some(hash_result))
 }
 
+fn find_hash(input: &PathBuf) -> Option<(HashAlg, &str, Vec<u8>)> {
+    for (alg, hash_file) in HASH_FILES {
+        let hash_filepath = input.parent()?.join(hash_file);
+        if let Ok(true) = fs::exists(&hash_filepath) {
+            if let Some(expected_hash) = parse_hashfile(
+                &mut BufReader::new(File::open(hash_filepath).ok()?),
+                input.file_name()?.to_str()?,
+            ) {
+                return Some((alg, hash_file, expected_hash));
+            }
+        }
+    }
+
+    None
+}
+
+fn parse_hashfile(hash_file: &mut impl BufRead, input_file: &str) -> Option<Vec<u8>> {
```

Comment:
probably best that this return a `Result` rather than using `.ok()` in the implementation
```suggestion
fn parse_hashfile(hash_file: &mut impl BufRead, input_file: &str) -> std::io::Result<Option<Vec<u8>>> {
```

Reply:
The return value of `base16::decode` is `Result<Vec<u8>, DecodeError>`, which is incompatible with `std::io::Result`.

```
error[E0308]: mismatched types
   --> src/ui/simple_ui/ask_hash.rs:120:24
    |
116 | fn parse_hashfile(hash_file: impl BufRead, input_file: &str) -> std::io::Result<Option<Vec<u8>>> {
    |                                                                 -------------------------------- expected `Result<std::option::Option<Vec<u8>>, futures_io::Error>` because of return type
...
120 |                 return base16::decode(hash.as_bytes())
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Result<Option<Vec<u8>>, Error>`, found `Result<Vec<u8>, DecodeError>`
    |
    = note: expected enum `Result<std::option::Option<Vec<_>>, futures_io::Error>`
               found enum `Result<Vec<_>, base16::DecodeError>`
```

Reply:
I use `anyhow::Result` to solve this as `DecodeError` is incompatible with `std::io::Error`

Reply:
yes, that works